### DISCUSSION TITLE
Display current font_size in status

### DISCUSSION THREAD

**Discussion by bugabinga** [wezterm/wezterm#5495] [Q&A] [answered]
### What Operating System(s) are you running on?

Windows, Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

Gnome

### WezTerm version

20240203-110809-5046fc22

### Ask your question!

Hi folks!

I have this little issue with displaying the current font size in the (right) status.
It shows up correctly initially, but does not get updated, when font sizes change interactively (via key binds).
Here is my update function:

```lua
local wez = require 'wezterm'

local DELIMITER = ' · '

local zoomed = function(pane)
	local tab = pane:tab()
	local any_is_zoomed = false
	if tab then
		for _, info in ipairs(tab:panes_with_info()) do
			if info.is_zoomed then
				any_is_zoomed = true
			end
		end
	end
	return any_is_zoomed and '🔎 Zoomed Pane' or nil
end

-- Show which key table is active in the status area
wez.on('update-status', function(window, pane)
	local key_table = window:active_key_table()
	local current_workspace = window:active_workspace()
	local zoomed_state = zoomed(pane)
	local leader = nil
	if window:leader_is_active() then
		leader = 'LEADER'
	end
	local fonts = {}
	local config = window:effective_config()
	local font_size = config.font_size
	for _, font in pairs(config.font.font) do
		table.insert(fonts, string.format('%s:%s', font.family, font_size))
	end
	local font = wez.format {
		{ Text = ' Font ' },
		{ Attribute = { Intensity = 'Bold' } },
		{ Text = table.concat(fonts, ', ') },
		'ResetAttributes'
	}

	local status = { ' ' }
	if leader then table.insert(status, leader) end
	if key_table then table.insert(status, key_table) end
	if zoomed_state then table.insert(status, zoomed_state) end
	if current_workspace then table.insert(status, current_workspace) end
	if font then table.insert(status, font) end

	window:set_right_status(table.concat(status, DELIMITER))
	window:set_left_status('   ')
end)
```

Any idea what I might be doing wrong?

Hope you’ll have a wonderful day!

**Comment by pkazmier** [comment, marked as answer]
It actually does work, but when you use the standard bindings to change font size, you are changing a zoom level vs the actual font size. Here is your code (just the font part) working as I change font sizes explicitly:

https://github.com/wez/wezterm/assets/747855/4a4ea716-4c18-4636-b007-7fd4cfc4e684

I use my [config-selector](https://github.com/wez/wezterm/discussions/5435) to change fonts and font sizes.

**Reply by pkazmier** [reply]
If you want `SUPER =` and `SUPER -` to change font size explicitly instead of by 10%, you could do something like this:

```lua
wezterm.on("my-inc-font-size", function(window)
  local size = window:effective_config().font_size + 1
  local overrides = window:get_config_overrides() or {}
  overrides.font_size = size
  window:set_config_overrides(overrides)
end)

wezterm.on("my-dec-font-size", function(window)
  local size = window:effective_config().font_size - 1
  local overrides = window:get_config_overrides() or {}
  overrides.font_size = size
  window:set_config_overrides(overrides)
end)

config.keys = {
  { key = "=",      mods = "SUPER",  action = act.EmitEvent "my-inc-font-size" },
  { key = "-",      mods = "SUPER",  action = act.EmitEvent "my-dec-font-size" },
}
```

I just tested and it works as you want it to work.

**Reply by bugabinga** [reply]
ah so cool!
and you just taught me how to implement custom actions!

very much appreciated

**Reply by pkazmier** [reply]
You're welcome! And you just taught me that I can have the font and size in my status bar, so thank you! My finished product below:

<img width="946" alt="Screenshot 2024-06-01 at 3 22 01 PM" src="https://github.com/wez/wezterm/assets/747855/2f521164-9593-40e7-9b67-e1af7e96f018">

**Reply by bugabinga** [reply]
<details>
<summary>random nerdfonts</summary>

~~~lua
	local nerdfonts = io.open(wez.config_dir .. '/nerdfonts', 'r')
	if nerdfonts then
		for font_name in nerdfonts:lines() do
			if not font_name:match '^#.*' then
				table.insert(font_names, font_name)
			end
		end
		local random_font = font_names[math.random(#font_names)]
		font = wez.font(random_font)
	end

~~~

the `nerdfonts` file is just a text file with one font name per line.
</details>


i am still searching for my favourite coding font ;)

judging from your screenshot and your config-selector thing, you have already found some you are happy with.
comic code is hilarious. and surprisingly legible ...

**Reply by pkazmier** [reply]
Yes, Comic Code was a recent addition this week to my collection. Normally, I wouldn't consider a font like this, but as you point out, it is extremely legible even at smaller sizes. It's quickly become a favorite. My all-time favorite is PragmataPro (I don't mind paying for fonts). Here is the list I've been using lately:

https://github.com/pkazmier/wezterm-config/tree/main/fonts
